service/service.py

```diff
@@ -1,5 +1,122 @@
+"""
+Supplier Store Service
+
+Paths:
+------
+GET /suppliers - Returns a list all of the Suppliers
+GET /suppliers/{id} - Returns the Supplier with a given id number
+POST /suppliers - creates a new Supplier record in the database
+PUT /suppliers/{id} - updates a Supplier record in the database
+DELETE /suppliers/{id} - deletes a Supplier record in the database
+TODO Query and Action Paths
+"""
+
+import sys
+import logging
+from flask import jsonify, request, json, url_for, make_response, abort
+from flask_api import status    # HTTP Status Codes
+from werkzeug.exceptions import NotFound
+from service.models import Supplier
 from . import app
 
+# Error handlers require app to be initialized so we must import
+# them only after we have initialized the Flask app instance
+import service.error_handlers
+
 @app.route('/')
 def hello():
-    return "Hello Supplier!"
\ No newline at end of file
+    return "Hello Supplier!"
+
+
+######################################################################
+# CREATE A NEW SUPPLIER
+######################################################################
+@app.route('/suppliers', methods=['POST'])
+def create_suppliers():
+    """
+    Creates a Supplier
+    This endpoint will create a Supplier based the data in the body that is posted
+    """
+    app.logger.info('Request to Create a Supplier...')
+    data = {}
+    # Check for form submission data
+    if request.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
+        app.logger.info('Getting data from form submit')
+        data = {
+            # "id": request.form['id'],
```

Comment:
Aren't we adding the ID of the supplier?

Reply:
Originally in the template given by the professor (Pets also have id) it does not have this attribute in this part. I am also confused (if you see the model definination/serialze/deserialze ) so I commented this sentence cause if later people tightly follow the template, might have some issue with this part.

Reply:
but let me fix this and add the id now since it is more in accord with the real situation. the database also produces an "_id" attribute.